algorithms/src/Kokkos_Random.hpp

```diff
@@ -903,24 +905,31 @@ class Random_XorShift64_Pool {
 #else
   Random_XorShift64_Pool()   = default;
 #endif
-  Random_XorShift64_Pool(uint64_t seed) {
-    num_states_ = 0;
 
-    init(seed, execution_space().concurrency());
+  Random_XorShift64_Pool(uint64_t seed)
+      : Random_XorShift64_Pool(execution_space(), seed,
+                               Impl::RandomPoolConstructorIsAsync::no) {}
+
+  Random_XorShift64_Pool(execution_space const& exec, uint64_t seed,
+                         Impl::RandomPoolConstructorIsAsync is_async =
+                             Impl::RandomPoolConstructorIsAsync::yes)
+      : num_states_(exec.concurrency()) {
+    init(exec, seed);
+    if (!static_cast<bool>(is_async)) exec.fence();
```

Comment:
```suggestion
    if (!static_cast<bool>(is_async)) exec.fence("Kokkos::Random_XorShift64_Pool synchronous constructor");
```
(same comment for Random_XorShift1024)